app/assets/javascripts/govuk_publishing_components/components/layout-super-navigation-header.js

```diff
@@ -4,68 +4,210 @@ window.GOVUK = window.GOVUK || {}
+    setLabelToShow($button)
+  }
+  var show = function ($button, $menu) {
+    $button.setAttribute('aria-expanded', true)
+    $menu.removeAttribute('hidden')
+    setLabelToHide($button)
+  }
 
-    this.$menu = document.getElementById(this.$button.getAttribute('aria-controls'))
+  var toggle = function ($button, $menu) {
+    var isOpen = $button.getAttribute('aria-expanded') === 'true'
+    if (isOpen) {
+      hide($button, $menu)
+    } else {
+      show($button, $menu)
+    }
+  }
 
-    this.syncStatus()
+  // Clicking an element inside a `button` element causes the `event.target` to
+  // be the inside element, not the button. This can be taken care of by setting
+  // the CSS pointer-events to be none, but that doesn't work for older
+  // browsers, won't work for people with CSS turned off, or for people who are
+  // using CSS overrides.
+  //    This checks if the $element is the `elementType`; if it is, it gets
+  // returned; if not it recursively checks to see if the parent element is a
+  // `elementType`. This means that it can be used with `pointer-events: none`.
+  var closestParentIncluding = function ($element, elementType) {
+    if ($element.tagName.toLowerCase() === elementType.toLowerCase()) {
+      return $element
+    }
+    return closestParentIncluding($element.parentNode, elementType)
   }
 
-  SuperNavigationToggle.prototype.setupButton = function (target) {
-    var menuHeading = target.getAttribute('aria-labelledby')
-    var adjacentTo = target.querySelector('#' + menuHeading)
-    var buttonText = document.createTextNode(this.buttonText)
+  var closestPrevious = function ($element, elementType) {
+    if ($element === null) {
+      return false
+    }
 
-    var button = document.createElement('button')
-    button.type = 'button'
-    button.className = 'govuk-header__menu-button gem-c-layout-super-navigation-header__menu-button'
-    button.setAttribute('aria-controls', 'super-navigation-menu')
-    button.setAttribute('aria-label', this.showMenuText)
-    button.setAttribute('aria-expanded', false)
+    if ($element.nodeType === 1 && $element.tagName.toLowerCase() === elementType.toLowerCase()) {
+      return $element
+    }
+
+    return closestPrevious($element.previousSibling, elementType)
+  }
 
-    button.appendChild(buttonText)
-    adjacentTo.insertAdjacentElement('beforebegin', button)
+  // When moving from one screen size to another (eg mobile to desktop) we need
+  // to find out whether a submenu is open so the parent menu can be kept open.
+  var hasSubMenusOpen = function ($menu) {
+    return $menu.querySelectorAll('button[aria-expanded="true"]').length > 0
+  }
 
-    return button
+  // Returns what screen size the window is currently. Returns a string of
+  // either `desktop` or `mobile` so it can be interpolated to access the
+  // `data-toggle-{desktop|mobile}-group` attribute.
+  var windowSize = function () {
+    return window.innerWidth >= 768 ? 'desktop' : 'mobile'
```

Comment:
Could you use the `SETTINGS` constant above here?

Reply:
There was a reason why I did this, but I was daft enough to not comment it. Let me dig and see what happens.

Reply:
Turns out it works - changed in https://github.com/alphagov/govuk_publishing_components/pull/2223/commits/334b7459af79ca8ba71884671095d56c669cfd4b.